internal/storage/auth/auth.go

```diff
@@ -43,9 +43,10 @@ type Store interface {
 // CreateAuthenticationRequest is the argument passed when creating instances
 // of an Authentication on a target AuthenticationStore.
 type CreateAuthenticationRequest struct {
-	Method    auth.Method
-	ExpiresAt *timestamppb.Timestamp
-	Metadata  map[string]string
+	Method      auth.Method
+	ExpiresAt   *timestamppb.Timestamp
+	Metadata    map[string]string
+	ClientToken string
```

Comment:
A clarifying comment here might be nice. e.g.
```suggestion
	// ClientToken is an (optional) explicit client token to be associated with the authentication.
	// When it is not supplied a random token will be generated and returned instead.
	ClientToken string
```